main.go

```diff
@@ -116,3 +129,50 @@ Arguments:
 		helpTab,
 	)
 }
+
+// This section translates a windows path such as "C:\Users\user" to "/Users/user"
+// and removes the default paths for mingw and git4windows to issues when you
+// specify a file path to start code-server in.
+func gitbashWindowsDir(dir string) string {
+	if dir == "~" { //Special case
+		return "~/"
+	}
+	mountPoints := gitbashMountPointsAndHome()
+
+	// Apply mount points
+	absDir, _ := filepath.Abs(dir)
+	absDir = filepath.ToSlash(absDir)
+	for _, mp := range mountPoints {
+		if strings.HasPrefix(absDir, mp[0]) {
+			resolved := strings.Replace(absDir, mp[0], mp[1], 1)
+			flog.Info("Resolved windows path '%s' to '%s", dir, resolved)
+			return resolved
+		}
+	}
+	return dir
+}
+
+// This function returns an array with MINGW64 mount points including relative home dir
```

Comment:
`gitbashMountPoints` is a good enough function name for this. Since this probably works on platforms other than gitbash it might be better to just call this `mountPoints` or `systemMountPoints`

This comment doesn't adequately explain what the function does, it should be something along the lines of `gitbashMountPoints returns all mount point paths on the system using the mount command with the device name and the mount path. This is used on windows to expand file paths.`

Reply:
How about this? its a mix of the old one and the new one, while explaining while the error occurs
```go
// gitbashMountPoints returns all mount points in a msys2 system and then trunicates
// the mount point for `/` from the path provided (dir), this is done to fix an
// issue with how msys2 provides file paths to go, example. When you feed a filepath
// as user input to go, `command /opt` for example, msys2 will pass `C:\<path to msys2 directory>\opt`
// This causes an issue whith this program where when you run `sshcode.exe user@server /opt`,
// GO and msys2 will attempt to pass `C:\<path to msys2 directory>\opt` to the server,
```

Reply:
`res` doesn't get used anywhere else in the code and seems unneeded. All of the stuff to do with truncating seems unused to me.

```
gitbashMountPoints returns all mount point paths on the system using the mount command with the device name and the mount path. This is used on windows to expand file paths.

On msys2 environments, if you do `sshcode host /opt`, the program will receive `C:\path\to\msys2\opt` instead, so this is used to map windows paths to linux paths.
```